### PR TITLE
Fixes in github actions for clang-tidy and clang-format

### PR DESCRIPTION
**Pull request by vaithak**

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -26,14 +26,15 @@ Checks: >
   -readability-magic-numbers,
   -readability-named-parameter,
   -readability-function-cognitive-complexity,
+  -cppcoreguidelines-pro-bounds-pointer-arithmetic,
```
Comment (vaithak) [inline .clang-tidy:29]: Removed this check because clad::array deals with pointer arithmetic

Comment (vaithak) [reply .clang-tidy:29]: I don't think so; there is a similar discussion here: https://github.com/isocpp/CppCoreGuidelines/issues/862 for implementing a vector-like container.

Comment (vgvassilev) [reply .clang-tidy:29]: How about using that approach: https://clang.llvm.org/extra/clang-tidy/#suppressing-undesired-diagnostics

Comment (vaithak) [reply .clang-tidy]: Yup, it worked. I had no idea about this 😅

**Review by vgvassilev — APPROVED** [review]
There seem to be some unrelated failure with ubuntu22 and llvm 13. Otherwise, lgtm!

**Pull request merged**